compiler/rustc_hir_analysis/src/hir_ty_lowering/errors.rs

```diff
@@ -1215,7 +1216,111 @@ pub fn prohibit_assoc_item_binding(
         } else {
             None
         },
-    })
+    });
+
+    // Emit a suggestion to use the type arg directly
+    // if there's a generic param with a matching name
+    // otherwise suggest removal of type binding
+    if let Some((def_id, segment, _)) = segment
+        && segment.args().parenthesized == hir::GenericArgsParentheses::No
+        && let hir::TypeBindingKind::Equality { term } = binding.kind
+    {
+        // Suggests removal of the offending binding
+        let suggest_removal = |e: &mut Diag<'_>| {
```

Comment:
Not to self: I still need to review this function. I suspect it can be simplified by a lot but let's see.